api-list.go

```diff
@@ -208,12 +208,10 @@ func (c Client) listObjectsV2Query(bucketName, objectPrefix, continuationToken s
 		urlValues.Set("fetch-owner", "true")
 	}
 
-	// maxkeys should default to 1000 or less.
-	if maxkeys == 0 || maxkeys > 1000 {
```

Comment:
Should `maxKeys` be set to `1000` if it is `<=0`

Reply:
No, the intention is to avoid setting such that server implementations can send their default limit. Clients don't need to limit itself. 

It is a change that we shall do on the server as well where we can send 10000 entries instead of 1000